app/controllers/application_controller.rb

```diff
@@ -50,8 +50,9 @@ def authenticate
 
   def authenticate_token
     authenticate_with_http_token do |token, _options|
-      if (@current_user = User.find_by(api_key: token))
-        @current_user.identity = UserAttributes.new(username: @current_user.username)
+      digest = Keycard::DigestKey.new(key: token).digest
+      if (@current_user = User.find_by(api_key_digest: digest))
+        @current_user.identity = UserAttributes.new(username: @current_user.username).all
```

Comment:
Not sure why GitHub won't let me comment on the line above (44) that will need to change. We should just always go to Keycard, since we are planning to use the Authorization header. What I think that really means is that this method goes away, and the Keycard implementation get inlined in the `authenticate` method with this User find for setting the current_user.